src/Factory.php

```diff
@@ -40,10 +47,67 @@ public static function getImageClient($organization, $apiKey, $apiSecret, $baseU
      */
     public static function getUserClient($baseUrl = BaseClient::DEFAULT_API_BASE_URL)
     {
-        $client = new GuzzleClient([
-            'base_uri' => $baseUrl
-        ]);
+        $client = self::getGuzzleClient($baseUrl);
 
         return new UserClient($client);
     }
+
+    /**
+     * Returns a Guzzle client with a retry middleware
+     *
+     * @param string $baseUrl base url
+     * @return GuzzleClient   GuzzleClient to connect to the backend
+     */
+    private static function getGuzzleClient($baseUrl)
+    {
+        $handlerStack = HandlerStack::create(new CurlHandler());
```

Comment:
The CurlHandler should not be passed explicitly. Guzzle choses the best handler available based on the environment then.